pkg/providers/az-client.go

```diff
@@ -43,3 +46,36 @@ func (g *GraphServiceClient) GetApplicationObjectId(ctx context.Context, appId s
 	}
 	return *appObjectId, nil
 }
+
+type RoleAssignmentClient struct {
+	Client *armauthorization.RoleAssignmentsClient
+}
+
+type RoleAssignClient interface {
+	CreateRoleAssignment(ctx context.Context, objectId, roleId, scope, raUid string) error
+}
+
+var _ RoleAssignClient = &RoleAssignmentClient{}
+
+func (r *RoleAssignmentClient) CreateRoleAssignment(ctx context.Context, objectId, roleId, scope, raUid string) error {
+	log.Debug("Assigning contributor role to service principal...", "objectId", objectId, "role assignment UID", raUid, "scope", scope)
```

Comment:
we probably don't want to always say we are assigning contributor role to service principal if we take in the roleID as a param